### ISSUE TITLE
Unable to resolve reference rule

### ISSUE THREAD

**Issue by melroy89** [opnsense/core#7609]
**Describe the bug**

Firewall -> Diagnostics -> Sessions -> Look at the sessions has has a rule saying: "let out anything...".

Clicking on that link would open: `<opnsense-IP>/firewall_rule_lookup.php?rid=fae559338f65e11c53669fc3642c93c2`

This new page gives me an empty fully blank page?

**To Reproduce**

1. Go to Firewall->Diagnostics -> Sessions
2. Click on 'let out anythi...' rule text
3. Blank page

**Expected behavior**

No white page :).

**Screenshots**

![image](https://github.com/opnsense/core/assets/628926/4a02223a-f885-4938-98fb-dec087cd33a7)

**Relevant log files**

Not that I can think of?

**Environment**

OPNsense 24.1.9_4 (amd64).
Currently running in VM (Proxmox) for testing our firewall software

**Comment by melroy89** [comment]
> Thank you for creating an issue. Since the ticket doesn't seem to be using one of our templates, we're marking this issue as low priority until further notice.
> 
> For more information about the policies for this repository, please read https://github.com/opnsense/core/blob/master/CONTRIBUTING.md for further details.
> 
> The easiest option to gain traction is to close this ticket and open a new one using one of our templates.

Actually it's using the bug template. And removed one or 2 headings that weren't relevant. But OK.

**Comment by AdSchellevis** [comment]
when (internal) rules change after the state has been created, this happens. The machine just can't resolve the reference in its current set of rules. (easy to check by locating the same rule in the rules menu)

**Comment by melroy89** [comment]
I just notice the same problem is present in States as well as Sessions.

> when (internal) rules change after the state has been created, this happens.

But I still consider this a bug right? This is not expected behavior. I actually just installed OPNsense from scratch and this is what I get.

**Comment by AdSchellevis** [comment]
it's not a bug, at least not one solvable with the boundaries that exist (states may refer to rules that no longer exists, which is a fact of life, for good reasons by the way)

**Comment by darkk** [comment]
To the best of my understanding, the reported bug might be solvable, it's just a UI/UX one, not a rule lookup.

It affects a kinda special rule, the `'let out anything...'` one.

`Automatically generated rules` drop-down in the `Firewall: Rules: wan` manages to hide `fa-search` icon for this internal rule. My box has two rules having no icon:
 1. `Default deny / state violation rule `
 2. `let out anything from firewall host itself`

However, `Firewall: Diagnostics: States` and `Firewall: Diagnostics: Sessions` generate clickable links for these rules and that's the thing that confused the author.

So, it's no necessary a state referring to rules that no longer exists.

Also, af55c854bc was a relevant improvement. Extra `<h1>(Rule) Not found.</h1>` line might be less confusing than empty screen in case of `window.close()` being blocked by the browser for some reason.
